pkg/client/round_trippper.go

```diff
@@ -0,0 +1,32 @@
+package client
```

Comment:
we have multiple util dir, can you move this one file there or inside one of those? I don't foresee this dir to have many other files so it's kinda weird to have a top dir under pkg just having one funciton.

Reply:
I moved them to `pkg/utils/httpclient` in [this commit](https://github.com/Azure/fleet/pull/462/commits/d3cfe7eb3acc5746b6385fb31a6c09525482a693).